gateway-js/src/loadSupergraphSdlFromStorage.ts

```diff
@@ -67,10 +120,68 @@ export async function loadSupergraphSdlFromStorage({
         'content-type': 'application/json',
       },
     });
+
+    const oobResponse = await result.json();
```

Comment:
`oobResponse` vs `result` seems a bit overloaded in this context.  It seems plausible that both could even go away at this level (and maybe even the _outer_ `try` and `catch`?) by adopting an approach that `await`s the `fetch` with a `.then` which attempts to `result.json()` and throws unless `data?.reportError`?.  that could be followed by a chained `.catch(...)` and then perhaps the final `catch` + `throw` of nearly the same error (down below) could be removed entirely?

Reply:
@Y-Guo it's very possible one of my suggestions (re: preference) goes exactly against a change you've made per this recommendation from @abernix. If so, please feel even more free to disregard and leave as is.